pkg/security/resolvers/cgroup/model.go

```diff
@@ -0,0 +1,79 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux
+// +build linux
+
+package cgroup
+
+import (
+	"sync"
+
+	"github.com/DataDog/datadog-agent/pkg/security/utils"
+	"github.com/hashicorp/golang-lru/v2/simplelru"
+	"go.uber.org/atomic"
+)
+
+// Cgroup cgroup resolver cache entry
+type Cgroup struct {
+	sync.RWMutex
+	Deleted *atomic.Bool
+	ID      string
+	Tags    []string
+	PIDs    *simplelru.LRU[uint32, int8]
+}
+
+// NewCgroup returns a new instance of a Cgroup
+func NewCgroup(id string, pids ...uint32) (*Cgroup, error) {
+	pidsLRU, err := simplelru.NewLRU[uint32, int8](1000, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	newCGroup := Cgroup{
+		Deleted: atomic.NewBool(false),
+		ID:      id,
+		PIDs:    pidsLRU,
+	}
+
+	for _, pid := range pids {
+		newCGroup.PIDs.Add(pid, 0)
+	}
+	return &newCGroup, nil
+}
+
+// GetRootPIDs returns the list of root pids for the current workload
+func (cgce *Cgroup) GetRootPIDs() []uint32 {
```

Comment:
It's a nit but these aren't root pids anymore: these are all the pids inside this cgroup.